src/DataSource.ts

```diff
@@ -19,28 +19,31 @@ import {
   MyVariableQuery,
   MultiValueVariable,
   TextValuePair,
+  RequestFactory,
 } from './types';
 import { dateTime, MutableDataFrame, FieldType, DataFrame } from '@grafana/data';
 import { getTemplateSrv } from '@grafana/runtime';
-import _ from 'lodash';
 import { isEqual } from 'lodash';
 import { flatten, isRFC3339_ISO6801 } from './util';
+import { GraphQLObjectType, isObjectType } from 'graphql';
+import { Schema } from './schema';
 
 const supportedVariableTypes = ['constant', 'custom', 'query', 'textbox'];
 
-export class DataSource extends DataSourceApi<MyQuery, MyDataSourceOptions> {
-  basicAuth: string | undefined;
-  withCredentials: boolean | undefined;
-  url: string | undefined;
-
-  constructor(instanceSettings: DataSourceInstanceSettings<MyDataSourceOptions>, private backendSrv: any) {
-    super(instanceSettings);
-    this.basicAuth = instanceSettings.basicAuth;
-    this.withCredentials = instanceSettings.withCredentials;
-    this.url = instanceSettings.url;
+class _RequestFactory implements RequestFactory {
+  constructor(
+    private basicAuth: string | undefined,
+    private withCredentials: boolean | undefined,
+    private url: string,
+    private backendSrv: any
```

Comment:
should have a comma at the end here

Reply:
maybe not supposed to do that in typescript, I don't know

Reply:
If you add it, prettier will remove it.